features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/SessionReplayResource.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder
+
+import java.io.Serializable
+
+// An individual resource. It is used to describe binary representation of heavy resources such as images.
+internal data class SessionReplayResource(
```

Comment:
Because you provide your own implementation of the `equals` class this class needs to be unit tests. See the `BatchData` class I added recently and check the `BatchDataTest` class.